Source/WebKit/Platform/cocoa/WebPrivacyHelpers.h

```diff
@@ -87,6 +88,48 @@ class LinkDecorationFilteringController {
     WeakHashSet<LinkDecorationFilteringDataObserver> m_observers;
 };
 
+class StorageAccessUserAgentStringQuirkObserver : public RefCounted<StorageAccessUserAgentStringQuirkObserver>, public CanMakeWeakPtr<StorageAccessUserAgentStringQuirkObserver> {
+public:
+    ~StorageAccessUserAgentStringQuirkObserver() = default;
+
+private:
+    friend class StorageAccessUserAgentStringQuirkController;
+
+    StorageAccessUserAgentStringQuirkObserver(Function<void()>&& callback)
+        : m_callback { WTFMove(callback) }
+    {
+    }
+
+    static Ref<StorageAccessUserAgentStringQuirkObserver> create(Function<void()>&& callback)
```

Comment:
Why do we want this to be private? We usually have these public.

Reply:
I'm not sure, I was following the existing example. @charliewolfe @whsieh do you remember why these are private? (and below)